src/Moq/Mock.Generic.cs

```diff
@@ -164,7 +166,15 @@ private void CheckParameters()
 		public override bool CallBase
 		{
 			get => this.callBase;
-			set => this.callBase = value;
+			set
+			{
+				if (typeof(Delegate).IsAssignableFrom(this.MockedType))
```

Comment:
(As above:) Moq defines an [extension method `Type.IsDelegate()`](https://github.com/moq/moq4/blob/2c241fbd4599da0069668d07f69441d66a9f8687/src/Moq/Extensions.cs#L75-L81), could you please use it here?